### PR TITLE
Adding SONIC and ZKSYNC to Magpie Protocol Aggregator

### PR DESCRIPTION
**Pull request by andrewnguyen1997**

### CODE REVIEW

**Comment by llamabutler** [comment]
The magpie adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts aggregators magpie
    
    🦙 Running MAGPIE adapter 🦙
    ---------------------------------------------------
    Start Date:	Thu, 26 Dec 2024 00:00:00 GMT
    End Date:	Fri, 27 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    [Error: Request failed with status code 503] {
      url: 'https://prewimvk04.execute-api.us-west-1.amazonaws.com/prod/llama',
      method: 'POST',
      axiosError: 'Service Unavailable',
      chain: 'zksync'
    }

**Comment by Dongminh1997** [comment]
Fixed

```
[Error: Request failed with status code 503] {
  url: 'https://prewimvk04.execute-api.us-west-1.amazonaws.com/prod/llama',
  method: 'POST',
  axiosError: 'Service Unavailable',
  chain: 'zksync'
}
```

File: aggregators/magpie/index.ts [hunk 1]
```diff
@@ -92,6 +92,16 @@ const fetch = async (_t: number, _: ChainBlocks, {chain, startOfDay}: FetchOptio
           runAtCurrTime: true,
           start: '2022-09-08',
         },
+        [CHAIN.SONIC]: {
+          fetch: fetch,
+          runAtCurrTime: true,
+          start: 1735217146,
+        },
+        [CHAIN.ZKSYNC]: {
```
Comment (dtmkeng) [inline aggregators/magpie/index.ts]: is it zksync or zk sync era ?, if correct could you check error ?

**Review by dtmkeng — CHANGES_REQUESTED** [review]

**Comment by llamabutler** [comment]
The magpie adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts aggregators magpie
    
    🦙 Running MAGPIE adapter 🦙
    ---------------------------------------------------
    Start Date:	Sun, 29 Dec 2024 00:00:00 GMT
    End Date:	Mon, 30 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    ETHEREUM 👇
    Backfill start time: 8/9/2022
    NO METHODOLOGY SPECIFIED
    Daily volume: 16.33 k
    End timestamp: 1735430400 (2024-12-29T00:00:00.000Z)
    
    
    ARBITRUM 👇
    Backfill start time: 8/9/2022
    NO METHODOLOGY SPECIFIED
    Daily volume: 830.28 k
    End timestamp: 1735430400 (2024-12-29T00:00:00.000Z)
    
    
    POLYGON 👇
    Backfill start time: 8/9/2022
    NO METHODOLOGY SPECIFIED
    Daily volume: 9
    End timestamp: 1735430400 (2024-12-29T00:00:00.000Z)
    
    
    AVAX 👇
    Backfill start time: 8/9/2022
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1735430400 (2024-12-29T00:00:00.000Z)
    
    
    BSC 👇
    Backfill start time: 8/9/2022
    NO METHODOLOGY SPECIFIED
    Daily volume: 1.29 k
    End timestamp: 1735430400 (2024-12-29T00:00:00.000Z)
    
    
    OPTIMISM 👇
    Backfill start time: 8/9/2022
    NO METHODOLOGY SPECIFIED
    Daily volume: 301
    End timestamp: 1735430400 (2024-12-29T00:00:00.000Z)
    
    
    BASE 👇
    Backfill start time: 8/9/2022
    NO METHODOLOGY SPECIFIED
    Daily volume: 1.32 M
    End timestamp: 1735430400 (2024-12-29T00:00:00.000Z)
    
    
    SCROLL 👇
    Backfill start time: 8/9/2022
    NO METHODOLOGY SPECIFIED
    Daily volume: 376
    End timestamp: 1735430400 (2024-12-29T00:00:00.000Z)
    
    
    MANTA 👇
    Backfill start time: 8/9/2022
    NO METHODOLOGY SPECIFIED
    Daily volume: 25
    End timestamp: 1735430400 (2024-12-29T00:00:00.000Z)
    
    
    TAIKO 👇
    Backfill start time: 8/9/2022
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1735430400 (2024-12-29T00:00:00.000Z)
    
    
    POLYGON_ZKEVM 👇
    Backfill start time: 8/9/2022
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1735430400 (2024-12-29T00:00:00.000Z)
    
    
    BLAST 👇
    Backfill start time: 8/9/2022
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1735430400 (2024-12-29T00:00:00.000Z)
    
    
    METIS 👇
    Backfill start time: 8/9/2022
    NO METHODOLOGY SPECIFIED
    Daily volume: 954
    End timestamp: 1735430400 (2024-12-29T00:00:00.000Z)
    
    
    FANTOM 👇
    Backfill start time: 8/9/2022
    NO METHODOLOGY SPECIFIED
    Daily volume: 18
    End timestamp: 1735430400 (2024-12-29T00:00:00.000Z)
    
    
    SONIC 👇
    Backfill start time: 26/12/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 261.50 k
    End timestamp: 1735430400 (2024-12-29T00:00:00.000Z)
    
    
    ERA 👇
    Backfill start time: 8/9/2022
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1735430400 (2024-12-29T00:00:00.000Z)

File: aggregators/magpie/index.ts (hunk 1 above)
Comment (andrewnguyen1997) [reply aggregators/magpie/index.ts]: @dtmkeng I updated it

**Pull request merged**